lib/js/src/manager/screen/utils/VoiceCommand.js

```diff
@@ -0,0 +1,103 @@
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following
+* disclaimer in the documentation and/or other materials provided with the
+* distribution.
+*
+* Neither the name of the Livio Inc. nor the names of its contributors
+* may be used to endorse or promote products derived from this software
+* without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+class VoiceCommand {
+    /**
+     * NOTE: voiceCommands is poorly named. They are more like voice aliases, which map to a single voice command
+     * @param {String[]} voiceCommands
+     * @param {function} voiceCommandSelectionListener
+    */
+    constructor (voiceCommands = null, voiceCommandSelectionListener = null) {
+        // The strings the user can say to activate this voice command
+        this._voiceCommands = voiceCommands;
+        // The listener that will be called when the command is activated
+        this._voiceCommandSelectionListener = voiceCommandSelectionListener;
+        // Used Internally to identify the command
+        this._commandId = null;
+    }
+
+    /**
+     * The strings the user can say to activate this voice command
+     * @param {String[]} voiceCommands - the list of commands to send to the head unit
+     * @return {VoiceCommand}
+    */
+    setVoiceCommands (voiceCommands) {
+        this._voiceCommands = voiceCommands;
+        return this;
+    }
+
+    /**
+     * The strings the user can say to activate this voice command
+     * @return {String[]} - the List of voice commands
+    */
+    getVoiceCommands () {
+        return this._voiceCommands;
+    }
+
+    /**
+     * The listener that will be called when the command is activated
+     * @param {function} voiceCommandSelectionListener - the listener for this object
+     * @return {VoiceCommand}
+    */
+    setVoiceCommandSelectionListener (voiceCommandSelectionListener) {
+        this._voiceCommandSelectionListener = voiceCommandSelectionListener;
+        return this;
+    }
+
+    /**
+     * The listener that will be called when the command is activated
+     * @return {function} - the listener for this object
```

Comment:
Could be null (the default value).

```suggestion
     * @return {function|null} - the listener for this object
```